icepyx/core/auth.py

```diff
@@ -0,0 +1,118 @@
+import copy
+import datetime
+
+import earthaccess
+
+class AuthenticationError(Exception):
+    pass
+
+
+class EarthdataAuthMixin():
+    """
+    This class stores methods related to logging into Earthdata. It is inherited by
+    any other module that requires authentication.
+    """
+    def __init__(self, auth=None):
+        self._auth = copy.deepcopy(auth)
+        # initializatin of session and s3 creds is not allowed because those are generated
+        # from the auth object
+        self._session = None
+        self._s3login_credentials = None
+        self._s3_initial_ts = None  # timer for 1h expiration on s3 credentials
+
+    def __str__(self):
+        if self.session:
+            repr_string = "EarthdataAuth obj with session initialized"
+        else:
+            repr_string = "EarthdataAuth obj without session initialized"
+        return repr_string
+
+    @property
+    def auth(self):
+        # Only login the first time .auth is accessed
```

Comment:
It would be great if we could add some basic docstrings for these functions (and then also add this new module to the docs).